relay-server/src/utils/metrics_rate_limits.rs

```diff
@@ -73,6 +71,54 @@ pub fn extract_transaction_count(
     Some(TransactionCount { count, has_profile })
 }
 
+/// Extracts quota information from a list of metric buckets.
+pub fn extract_metric_quantities<'a, I, V>(buckets: I, mode: ExtractionMode) -> SourceQuantities
+where
+    I: IntoIterator<Item = V>,
+    BucketView<'a>: From<V>,
+{
+    let mut quantities = SourceQuantities::default();
+
+    for bucket in buckets {
+        quantities.buckets += 1;
+        if let Some(count) = count_metric_bucket(bucket.into(), mode) {
+            quantities.transactions += count.count;
+            if count.has_profile {
+                quantities.profiles += count.count;
+            }
+        }
+    }
+
+    quantities
+}
+
+pub fn reject_metrics(
+    addr: Addr<TrackOutcome>,
+    quantities: SourceQuantities,
+    scoping: Scoping,
+    outcome: Outcome,
+) {
+    let timestamp = Utc::now();
+
+    let categories = [
+        (DataCategory::Transaction, quantities.transactions as u32),
+        (DataCategory::Profile, quantities.profiles as u32),
+        (DataCategory::MetricBucket, quantities.buckets as u32),
+    ];
+
+    for (category, quantity) in categories {
+        addr.send(TrackOutcome {
```

Comment:
Shouldn't we check for `quantity > 0` here?

Reply:
Good catch, thank you!